.github/workflows/test-and-publish-nightly.yml

```diff
@@ -65,65 +136,58 @@ jobs:
           architecture: ${{ matrix.architecture }}
       - name: Checkout sources
         uses: actions/checkout@v4
-      - name: Build and test
-        id: build
+      - name: Set artifact name
+        id: set_artifact_name
+        run: echo "artifact_name=async-profiler-${GITHUB_SHA:0:7}-${{ matrix.asprof-binaries-job || matrix.runson.display }}" >> $GITHUB_OUTPUT
+        shell: bash
+        env:
+          GITHUB_SHA: ${{ github.sha }}
+      - name: Download async-profiler release artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ steps.set_artifact_name.outputs.artifact_name }}
+          path: async_profiler_release
+      - name: Download async-profiler JAR artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: async-profiler-jars
+          path: jar_artifacts
+      - name: Extract async-profiler artifact
+        id: extract_artifact
         run: |
-          HASH=$(echo ${{ github.sha }} | cut -c-7)
+          release_archive=$(basename $(find async_profiler_release -type f -iname "async-profiler-*" ))
           case "${{ matrix.runson.name }}" in
             macos*)
-              make COMMIT_TAG=$HASH FAT_BINARY=true release test -j
+              unzip async_profiler_release/$release_archive
             ;;
             *)
-              make COMMIT_TAG=$HASH CC=/usr/local/musl/bin/musl-gcc release test -j
+              tar xvf async_profiler_release/$release_archive
             ;;
           esac
-          echo "archive=$(basename $(find . -type f -iname "async-profiler-*" ))" >> $GITHUB_OUTPUT
-      - name: Coverage
-        id: coverage
+          echo "jars_directory=jar_artifacts" >> $GITHUB_OUTPUT
+          echo "release_directory=$(basename $(find . -type d -iname "async-profiler-*" ))" >> $GITHUB_OUTPUT
+      - name: Run integration tests
         run: |
-          HASH=$(echo ${{ github.sha }} | cut -c-7)
-          case "${{ matrix.runson.name }}" in
-            macos*)
-              brew install gcovr
-              make COMMIT_TAG=$HASH FAT_BINARY=true coverage -j
-            ;;
-            *)
-              make COMMIT_TAG=$HASH CC=/usr/local/musl/bin/musl-gcc coverage -j
-            ;;
-          esac
-      - name: Upload test logs
+          mkdir -p build/jar
+          cp ${{ steps.extract_artifact.outputs.jars_directory }}/* build/jar
+          make build/test.jar
```

Comment:
Why are we building the jars here, shouldn't we take the ones built in the "build" steps?

Reply:
Let's keep it this way, makes sense.